common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -39,14 +39,16 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
+  class AmpExternalVideo(val videoId: String, val elementType: String) {}
+
   object AmpExternalVideo {
-    def getElementTypeByUrl(videoUrl: String) : String = {
-      val youtubePattern = ".*youtube.com.*".r
-      val vimeoPattern = ".*vimeo.com.*".r
+    def getAmpExternalVideoByUrl(videoUrl: String) : Option[AmpExternalVideo] = {
+      val youtubePattern = ".*https?://www.youtube.com/watch\\?v=(.+).*".r
+      val vimeoPattern = ".*https?://vimeo.com/(\\d+).*".r
```

Comment:
Not sure this regex is quite right

- Probably start with `^` instead of `.*`
- I think there are a few `/` and `.` that need escaping
- I think the query param ? should be `\?`, it looks like it's escaped twice
- Can we capture the youtube video id with `.*v=([\d\w]+)&?.*` - otherwise we might capture other query params before or after video id.

I might be off mark on some of this as I don't have good knowledge of youtube/vimeo urls and their edge cases.  Let me know what you think.